modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteTableModify.java

```diff
@@ -96,4 +108,64 @@ public IgniteTableModify(RelInput input) {
         return new IgniteTableModify(cluster, getTraitSet(), getTable(), sole(inputs),
             getOperation(), getUpdateColumnList(), getSourceExpressionList(), isFlattened());
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Pair<RelTraitSet, List<RelTraitSet>>> deriveRewindability(RelTraitSet nodeTraits,
+                                                                          List<RelTraitSet> inTraits) {
+        return ImmutableList.of(Pair.of(nodeTraits.replace(TraitUtils.rewindability(inTraits.get(0))),
+                inTraits));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Pair<RelTraitSet, List<RelTraitSet>>> deriveDistribution(RelTraitSet nodeTraits,
+                                                                         List<RelTraitSet> inTraits) {
+        return ImmutableList.of(Pair.of(nodeTraits.replace(TraitUtils.distribution(inTraits.get(0))),
+                inTraits));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Pair<RelTraitSet, List<RelTraitSet>>> deriveCollation(RelTraitSet nodeTraits,
+                                                                      List<RelTraitSet> inTraits) {
+        return ImmutableList.of(Pair.of(nodeTraits.replace(TraitUtils.collation(inTraits.get(0))),
+                inTraits));
+    }
+
+    /**
+     *
+     */
+    @Override
+    public Pair<RelTraitSet, List<RelTraitSet>> passThroughCorrelation(RelTraitSet nodeTraits,
+                                                                       List<RelTraitSet> inTraits) {
+        CorrelationTrait correlation = TraitUtils.correlation(nodeTraits);
+
+        return Pair.of(nodeTraits, ImmutableList.of(inTraits.get(0).replace(correlation)));
+    }
+
+    /**
+     *
+     */
+    @Override
+    public List<Pair<RelTraitSet, List<RelTraitSet>>> deriveCorrelation(RelTraitSet nodeTraits,
+                                                                        List<RelTraitSet> inTraits) {
+        Set<CorrelationId> corrIds = RexUtils.extractCorrelationIds(getSourceExpressionList());
```

Comment:
Currently IgniteTableModify doesn't introduce any correlations, so we should propagate its child's correlation only